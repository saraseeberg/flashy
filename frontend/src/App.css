#root {
  max-width: 1920px;
  width: 100%;
  text-align: center;
}
