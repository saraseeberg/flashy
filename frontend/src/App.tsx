import { Outlet } from 'react-router-dom';
import TopBar from './components/TopBar';
import { Box } from '@mui/material';

function App() {
  return (
    <Box
      sx={{
        height: '100vh',
        width: '100vw',
        overflowX: 'hidden',
        textAlign: 'center',
      }}
    >
      <TopBar />
      <Box
        sx={{
          display: 'flex',
          flexDirection: 'column',
        }}
      >
        <Outlet />
      </Box>
    </Box>
  );
}

export default App;
