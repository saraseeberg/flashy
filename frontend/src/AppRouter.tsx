import {
  BrowserRouter as Router,
  Routes,
  Route,
  Navigate,
} from 'react-router-dom';
import Login from './pages/LoginPage';
import Register from './pages/RegisterPage';
import Dashboard from './pages/DashboardPage';
import { useAuth } from './hooks/useAuth';
import CreateSet from './pages/CreateLearningsetPage';
import EditSet from './pages/EditLearningSetPage';
import ViewCards from './pages/ViewCardsPage';
import NotFound from './pages/NotFoundPage';
import Settings from './pages/SettingsPage';
import App from './App';
import AdminPage from './pages/AdminPage';
import ErrorPopup from './components/ErrorPopup';
import CreateAdminUserPage from './pages/CreateAdminUserPage';
import useUserRole from './hooks/useRole';

function AppRouter() {
  const { isAuthenticated } = useAuth();
  const userRole = useUserRole();

  return (
    <Router>
      <Routes>
        <Route path="/" element={<App />}>
          <Route path="/" element={<Navigate to="login" />} />
          <Route path="login" element={<Login />} />
          <Route path="register" element={<Register />} />
          <Route
            path="dashboard"
            element={isAuthenticated ? <Dashboard /> : <Navigate to="/login" />}
          />
          <Route
            path="create-set"
            element={isAuthenticated ? <CreateSet /> : <Navigate to="/login" />}
          />
          <Route
            path="edit-set/:setId"
            element={isAuthenticated ? <EditSet /> : <Navigate to="/login" />}
          />
          <Route
            path="viewcards/:setId"
            element={isAuthenticated ? <ViewCards /> : <Navigate to="/login" />}
          />
          <Route
            path="adminpage"
            element={
              (isAuthenticated && userRole === 'superadmin') ||
              (isAuthenticated && userRole === 'admin') ? (
                <AdminPage />
              ) : (
                <ErrorPopup
                  open={true}
                  onClose={() => <Navigate to="/dashboard" />}
                />
              )
            }
          />
          <Route
            path="create-adminUser"
            element={
              isAuthenticated && userRole === 'superadmin' ? (
                <CreateAdminUserPage />
              ) : (
                <ErrorPopup
                  open={true}
                  onClose={() => <Navigate to="/adminpage" />}
                />
              )
            }
          />
          <Route path="settings" element={<Settings />} />
          <Route path="*" element={<NotFound />} />
        </Route>
      </Routes>
    </Router>
  );
}

export default AppRouter;
