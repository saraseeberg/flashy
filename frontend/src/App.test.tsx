import { render, screen } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';
import AppRouter from './AppRouter';

vi.mock('./hooks/useAuth', () => ({
  useAuth: vi.fn().mockReturnValue({
    isAuthenticated: true,
    login: vi.fn(),
    logout: vi.fn(),
  }),
}));

describe('App', () => {
  it('should render without crashing', () => {
    const { container } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});

describe('App routing', () => {
  it('should render the login page when not authenticated', async () => {
    render(<AppRouter />);
    expect(await screen.findByText(/Log in/i)).toBeInTheDocument();
  });
});
