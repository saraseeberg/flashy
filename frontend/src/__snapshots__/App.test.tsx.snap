// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render without crashing 1`] = `
<div>
  <div
    class="MuiBox-root css-1my4ydd"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-8bjgx3-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1x5wi8o-MuiToolbar-root"
      >
        <div
          class="MuiBox-root css-70qvj9"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-dj6h1h-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <img
              alt="logo"
              src="/src/assets/flashy-logo.svg"
              style="max-width: 50px;"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap css-1961ak2-MuiTypography-root"
            style="color: black; padding-left: 10px;"
          >
            FLASHY
          </h5>
        </div>
        <div
          class="MuiBox-root css-70qvj9"
        >
          <div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SupervisorAccountIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.5 12c1.38 0 2.49-1.12 2.49-2.5S17.88 7 16.5 7C15.12 7 14 8.12 14 9.5s1.12 2.5 2.5 2.5M9 11c1.66 0 2.99-1.34 2.99-3S10.66 5 9 5C7.34 5 6 6.34 6 8s1.34 3 3 3m7.5 3c-1.83 0-5.5.92-5.5 2.75V19h11v-2.25c0-1.83-3.67-2.75-5.5-2.75M9 13c-2.33 0-7 1.17-7 3.5V19h7v-2.25c0-.85.33-2.34 2.37-3.47C10.5 13.1 9.66 13 9 13"
                  />
                </svg>
              </span>
              Admin Page
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-q4d9s0-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiBox-root css-j7qwjs"
    />
  </div>
</div>
`;
